src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -348,6 +348,18 @@ public function orWhereDoesntHaveMorph($relation, $types, Closure $callback = nu
         return $this->doesntHaveMorph($relation, $types, 'or', $callback);
     }
 
+    /**
+     * @param array $relations
+     * @param Closure|null $callback
+     * @return \Illuminate\Database\Eloquent\Builder|static
+     */
+    public function whereDoesntHaveAny($relations, Closure $callback = null)
+    {
+        foreach ($relations as $relation) {
+            return $this->doesntHave($relation);
```

Comment:
This will only be executed for the first relation of the array. Also the `$callback` argument is never used.

Reply:
Yeah sorry i created the pr as a draft cause i'm not sure how to run the tests locally otherwise i'd have done this locally before creating 🙈.

Thanks for the pointers!